src/odemis/driver/technolution.py

```diff
@@ -0,0 +1,544 @@
+        self.scanGain = model.TupleContinuous((0.0, 0.0), range=((-10.0, -10.0), (10.0, 10.0)), unit='V,V')
+
+
+class MPPC(model.Detector):
+    def __init__(self, name, role, parent, **kwargs):
+        super(MPPC, self).__init__(name, role, **kwargs)
+        self.parent = parent
+
+        if not (hasattr(self.parent, "_MirrorDescanner")) or not (hasattr(self.parent, "_EBeamScanner")):
+            ConnectionError("Not all required children are defined (maybe definition order is not correct)")
+
+        self._field_data = FieldMetaData(None, None)  # Current field_data
+
+        # VA defauls values
+        self._shape = model.TupleVA((8, 8, 65536), unit="cell-images and depth")
+        self.path = model.StringVA('~/Pictures')
+        self.dataContent = model.StringVA('empty')
+        self.filename = model.StringVA(time.strftime("default-%H-%M-%S-%Y-%m-%d"))
+        self._server_URL = self.parent.server_URL
+        self.externalStorageURL_host = model.StringVA('host')
+        #NOTE: Do not write real username/password here since this is published on github in plaintext!
+        self.externalStorageURL_user = model.StringVA('user')
+        self.externalStorageURL_password = model.StringVA('password')
+        self.acqDelay = model.FloatVA(2.0, unit='s', setter=self._setAcqDelay)
+
+        # Cell acquisition parameters
+        #TODO K.K. Perform type check on intput into the VA's floats and ints
+        self.cellTranslation = model.ListVA([[[50, 50]] * self._shape.value[0]] * self._shape.value[1])
+        # TODO K.K. Put a check according to this input in the assemble_mega_field_check whih includes the counting up
+        #  per list
+        self.cellTranslation = model.ListVA([[[10 + j, 20 + j] for j in range(i,i+self._shape.value[0])]
+                                             for i in range(0,self._shape.value[1]*self._shape.value[0],self._shape.value[0])])
+        self.celldarkOffset = model.ListVA([[0] * self._shape.value[0]] * self._shape.value[1])
+        self.celldigitalGain = model.ListVA([[1.2] * self._shape.value[0]] * self._shape.value[1])
+        self.celldigitalGain = model.ListVA([[j for j in range(i, i+self._shape.value[0])]
+                                        for i in range(0,self._shape.value[1]*self._shape.value[0], self._shape.value[0])])
+        self.cellCompleteResolution = model.ResolutionVA((800, 800), ((10, 10), (900, 900)))
+
+        # Gather metadata from Siblings and own _meta_data
+        self.md_devices = [self._metadata, self.parent._MirrorDescanner._metadata, self.parent._EBeamScanner._metadata]
+        self._metadata[model.MD_HW_NAME] = "MPPC"
+        self._metadata[model.MD_SW_VERSION] = self._swVersion
+        self._metadata[model.MD_ACQ_DATE] = int(time.time())  # Time since Epoch
+        self._metadata[model.MD_POS] = (0.0, 0.0)  # TODO implement in next API
+
+        self.acq_queue = queue.Queue()  # acquisition queue with steps that need to be executed
```

Comment:
We discussed this a while ago, queue.Queue() is more suitable for use in applications with threading etc and has methods which allow waiting until a new item is available.
https://stackoverflow.com/questions/717148/queue-queue-vs-collections-deque

Reply:
Yes we discussed before and I also read the stackoverflow discussion. I would like to learn more about when to chose queue or deque. In the stackoverflow discussion I cannot find a consensus if deque is thread safe or not (or only certain methods). Also queue seems to rely on deque. However, it looks like people agree that deque is faster than queue. It looks like indeed that queue has a blocking option and deque doesn't. Both seem to provide some order. Maybe it would be nice to have knowledge transfer about that.

Reply:
The two types of objects are thread-safe, both return values in FIFO order. However, they have a very big difference, as Kornee noted: Queue can block. For message passing between threads, this is what you need. This allows the "worker" thread to wait until a new event happens (or a new work is pushed, depending on the design pattern).